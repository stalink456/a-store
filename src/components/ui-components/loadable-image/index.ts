export { LoadableImage } from './loadable-image';
