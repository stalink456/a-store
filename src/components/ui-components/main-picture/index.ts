export { MainPicture } from "./main-picture";
