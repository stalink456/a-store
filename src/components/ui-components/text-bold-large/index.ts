export { TextBoldLarge } from "./text-bold-large";
