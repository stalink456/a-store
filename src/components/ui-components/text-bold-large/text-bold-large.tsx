import React from "react";
import { Typography } from "@alfalab/core-components/typography";
import { TextType } from "types";

export const TextBoldLarge: React.FC<TextType> = React.memo(({ text }) => {
  return (
    <React.Fragment>
      <Typography.Text tag="span" weight="bold" view="primary-large">
        {text}
      </Typography.Text>
    </React.Fragment>
  );
});
