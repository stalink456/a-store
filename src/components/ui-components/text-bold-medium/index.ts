export { TextBoldMedium } from "./text-bold-medium";
