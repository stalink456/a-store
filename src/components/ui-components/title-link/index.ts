export { TitleLink } from './title-link';
