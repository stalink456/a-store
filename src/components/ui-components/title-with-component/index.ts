export { TitleWithComponent } from "./title-with-component";
