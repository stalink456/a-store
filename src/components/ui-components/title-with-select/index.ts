export { TitleWithSelect } from './title-with-select';
