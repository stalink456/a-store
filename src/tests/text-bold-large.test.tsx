import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { TextBoldLarge } from 'components/ui-components/text-bold-large';

describe('TextBoldLarge component', () => {
  it('should render TextBoldLarge component', () => {
    const { baseElement } = render(<TextBoldLarge text='A-Store' />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should render TextBoldLarge component with title', async () => {
    render(<TextBoldLarge text='A-Store' />);

    expect(await screen.findByText('A-Store')).toBeInTheDocument();
  });

  it("shouldn't render TextBoldLarge component with non-existent text", async () => {
    render(<TextBoldLarge text='A-Store' />);

    expect(screen.queryByText('non-existent text')).not.toBeInTheDocument();
  });
});
