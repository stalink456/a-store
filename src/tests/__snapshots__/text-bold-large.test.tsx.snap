// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBoldLarge component should render TextBoldLarge component 1`] = `
<body>
  <div>
    <span
      class="typography__primary-large_c16y5 typography__bold_c16y5"
    >
      A-Store
    </span>
  </div>
</body>
`;
